compiler/src/main/scala/org/scalajs/nscplugin/GenJSCode.scala

```diff
@@ -6472,8 +6477,9 @@ abstract class GenJSCode[G <: Global with Singleton](val global: G)
             yield (formal.ref, param.tpe)
         }.map((ensureBoxed _).tupled)
 
-        val call = js.JSFunctionApply(
-            js.Select(js.This()(thisType), fFieldIdent)(jstpe.AnyType),
+        val call = js.ApplyTypedClosure(
+            js.ApplyFlags.empty,
+            js.Select(js.This()(thisType), fFieldIdent)(closureType),
```

Comment:
I'm trying to understand why we cannot use `NewLambda` here as well. Is it because of the bridges?

Reply:
As long as there aren't any bridges needed, we definitely can use `NewLambda` here. I haven't implemented it yet. That's what I was referring to in

> It is also generalizable to SAM types such as `j.u.Comparator`, although I did not implement that yet.

When there are bridges, however, we won't be able to do that. We could if we made `NewLambda` more powerful, to some extent. There are cases where even on the JVM they need to add "adapted" methods to satisfy the constraints of `LambdaMetaFactory`. AFAICT, the use cases for bridges are rare enough that adding *partial* support for more of them is not worth it.